src/legacy/core_plugins/kibana/public/context/api/context.js

```diff
@@ -54,110 +61,93 @@ function fetchContextProvider(indexPatterns, Private) {
   const SearchSource = Private(SearchSourceProvider);
 
   return {
-    fetchPredecessors,
-    fetchSuccessors,
+    // @ts-ignore / for testing
+    fetchPredecessors: (...args) => fetchSurroundingDocs('predecessors', ...args),
+    // @ts-ignore / for testing
+    fetchSuccessors: (...args) => fetchSurroundingDocs('successors', ...args),
+    fetchSurroundingDocs,
   };
 
-  async function fetchSuccessors(
+  /**
+   * Fetch successor or predecessor documents of a given anchor document
+   *
+   * @param {SurroundingDocType} type - `successors` or `predecessors`
+   * @param {string} indexPatternId
+   * @param {string} timeFieldName - name of the timefield, that's sorted on
+   * @param {SortDirection} timeFieldSortDir - direction of sorting
+   * @param {string} timeFieldIsoValue - value of the anchors timefield in ISO format
+   * @param {number} timeFieldNumValue - value of the anchors timefield in numeric format (invalid for nanos)
+   * @param {string} tieBreakerField - name of 2nd param for sorting
+   * @param {string} tieBreakerValue - value of 2nd param for sorting
+   * @param {number} size - number of records to retrieve
+   * @param {any[]} filters - to apply in the elastic query
+   * @returns {Promise<object[]>}
+   */
+  async function fetchSurroundingDocs(
+    type,
     indexPatternId,
-    timeField,
-    timeSortDirection,
-    timeValue,
+    timeFieldName,
+    timeFieldSortDir,
+    timeFieldIsoValue,
+    timeFieldNumValue,
     tieBreakerField,
     tieBreakerValue,
     size,
     filters
   ) {
-    const searchSource = await createSearchSource(indexPatternId, filters);
-    const offsetSign = timeSortDirection === 'asc' ? 1 : -1;
-
-    // ending with `null` opens the last interval
-    const intervals = asPairs([...LOOKUP_OFFSETS.map(offset => timeValue + offset * offsetSign), null]);
-
-    let successors = [];
-    for (const [startTimeValue, endTimeValue] of intervals) {
-      const remainingSize = size - successors.length;
-
-      if (remainingSize <= 0) {
-        break;
-      }
-
-      const [afterTimeValue, afterTieBreakerValue] = successors.length > 0
-        ? successors[successors.length - 1].sort
-        : [timeValue, tieBreakerValue];
-
-      const hits = await fetchHitsInInterval(
-        searchSource,
-        timeField,
-        timeSortDirection,
-        startTimeValue,
-        endTimeValue,
-        afterTimeValue,
-        tieBreakerField,
-        afterTieBreakerValue,
-        remainingSize
-      );
-
-      successors = [...successors, ...hits];
-    }
+    const indexPattern = await indexPatterns.get(indexPatternId);
+    const searchSource = await createSearchSource(indexPattern, filters);
+    const sortDir = type === 'successors' ? timeFieldSortDir : reverseSortDirection(timeFieldSortDir);
+    const nanoSeconds = indexPattern.isTimeNanosBased() ? extractNanoSeconds(timeFieldIsoValue) : '';
```

Comment:
Here's the most important change, if the indexPattern is based on date_nanos, this value has to be extracted of the ISO formatted timestamp, and that value is later on used to generate the `search_after` value, that Elasticsearch needs to return only records after a certain time. In inability of the browser handling `date_nanos` values (they are rounded automatically) led to the broken context view